windows/RNFS/RNFSManager.cpp

```diff
@@ -0,0 +1,1115 @@
+            StorageFolder src{ co_await StorageFolder::GetFolderFromPathAsync(item.Path()) };
+            StorageFolder dest{ co_await destFolder.CreateFolderAsync(item.Name(), CreationCollisionOption::OpenIfExists) };
+            copyFolderHelper(src, dest);
+        }
+    }
+
+    promise.Resolve();
+    co_return;
+}
+catch (const hresult_error& ex)
+{
+    // "Failed to copy file."
+    promise.Reject(winrt::to_string(ex.message()).c_str());
+}
+
+winrt::fire_and_forget RNFSManager::copyFolderHelper(
+    winrt::Windows::Storage::StorageFolder src,
+    winrt::Windows::Storage::StorageFolder dest) noexcept
+try
+{
+    auto items{ co_await src.GetItemsAsync() };
+    for (auto item : items)
+    {
+        if (item.IsOfType(StorageItemTypes::File))
+        {
+            StorageFile file{ co_await StorageFile::GetFileFromPathAsync(item.Path()) };
+            co_await file.CopyAsync(dest, file.Name(), NameCollisionOption::ReplaceExisting);
+        }
+        else if (item.IsOfType(StorageItemTypes::Folder))
+        {
+            StorageFolder srcFolder{ co_await StorageFolder::GetFolderFromPathAsync(item.Path()) };
+            StorageFolder destFolder{ co_await dest.CreateFolderAsync(item.Name(), CreationCollisionOption::OpenIfExists) };
+            copyFolderHelper(srcFolder, destFolder);
+        }
+    }
+}
+catch (...)
+{
+    co_return;
+}
+
+
+winrt::fire_and_forget RNFSManager::getFSInfo(RN::ReactPromise<RN::JSValueObject> promise) noexcept
+try
+{
+    auto localFolder{ Windows::Storage::ApplicationData::Current().LocalFolder() };
+    auto properties{ co_await localFolder.Properties().RetrievePropertiesAsync({L"System.FreeSpace", L"System.Capacity"}) };
+
+    RN::JSValueObject result;
+    result["freeSpace"] = unbox_value<uint64_t>(properties.Lookup(L"System.FreeSpace"));
+    result["totalSpace"] = unbox_value<uint64_t>(properties.Lookup(L"System.Capacity"));
+
+    promise.Resolve(result);
+}
+catch (const hresult_error& ex)
+{
+    // "Failed to retrieve file system info."
+    promise.Reject(winrt::to_string(ex.message()).c_str());
+}
+
+
+winrt::fire_and_forget RNFSManager::unlink(std::string filepath, RN::ReactPromise<void> promise) noexcept
+try
+{
+    size_t pathLength{ filepath.length() };
+
+    if (pathLength <= 0) {
+        promise.Reject("Invalid path.");
+    }
+    else {
+        bool hasTrailingSlash{ filepath[pathLength - 1] == '\\' || filepath[pathLength - 1] == '/' };
+        std::filesystem::path path(hasTrailingSlash ? filepath.substr(0, pathLength - 1) : filepath);
+        path.make_preferred();
+
+        StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(path.parent_path().wstring()) };
+        auto target{ co_await folder.GetItemAsync(path.filename().wstring()) };
+        co_await target.DeleteAsync();
+
+        promise.Resolve();
+    }
+}
+catch (const hresult_error& ex)
+{
+    hresult result{ ex.code() };
+    if (result == 0x80070002) // FileNotFoundException
```

Comment:
use HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)

Reply:
same in other places

Reply:
Replaced all occurances of that in my most recent commit.